cime_config/config_grids.xml

```diff
@@ -4153,6 +4153,11 @@
       <map name="ROF2OCN_LIQ_RMAPNAME">cpl/cpl6/map_r01_to_gx1v6_120711.nc</map>
     </gridmap>
 
+    <!--- ocean to river mapping files -->
+    <gridmap rof_grid="r05" ocn_grid="EC30to60E2r2">
+      <map name="OCN2ROF_SMAPNAME">cpl/cpl6/map_EC30to60E2r2_to_r05_nn_masked.210324.nc</map>
```

Comment:
@jonbob Thanks for the question. The mapping file remaps MPAS-O ssh to MOSART ocean cells and outlet cells. The ssh mapped to the outlet cells will be used as the downstream boundary of MOSART. Because there are usually no overlaps between the outlets and the ocean mesh, I use the ssh of the two nearest neighboring ocean cells for an outlet to calculate the mapped ssh instead of using the bilinear interpolation. The weight for each neighbor is the inverse of distance between the outlet and the ocean cell. This is equivalent to inverse distance weighting interpolation and the number of neighbored ocean cells can be arbitrary. I found 2 is adequate for this mesh. I agree with you that we may need to change the remapping method when the high-resolution unstructured mesh is available. The key is to make sure which ocean cells are mapped to the river outlets since they are not supposed to overlap.

Reply:
Is this mapping file made with a custom program like the r2o one?  Or did you use ncremap/Tempest/ESMF ?

Reply:
@rljacob, yes, the mapping file has to be made with a custom program. The MPAS-O ssh needs to be mapped to MOSART outlets. But in addition to the ocean outlets along the coastline, MOSART has dummy outlets on the land, in which discharge will be routed to the nearest ocean outlets. These land outlets have to be excluded from the remapping. So I use a custom remapping program that defines which outlets are included in a shapefile.

Reply:
@fdongyu - is that custom program documented and available for use?

Reply:
@jonbob The program is a Python script and has only been shared with ICoM team members. If it has to come with this PR, please let me know the best place to publish it (my own git repo or zenodo?). Thanks!

Reply:
@rljacob - what do you think? @fdongyu, I'm a bit curious how this is easier than mapping the ocean SSH to all the corresponding points on the global mosart grid with a more typical mapping, and then just using the outlet cell neighbors' value for SSH?

Reply:
@jonbob , thanks for the suggestion! I will do it that way. There would not be too much difference except that I will let each outlet search the nearest ocean ssh. Could you help me generate a mapping file for this example (EC30to60E2r2 to r05)? I tried the NCO remapping but got some errors.

Reply:
@fdongyu - I made a bilinear map, which we typically use for remapping states, for this and put it on lcrc at:
````
/lcrc/group/e3sm/data/inputdata/cpl/cpl6/map_EC30to60E2r2_to_r05_bilin.220727.nc
````
There is no masking done so you can hopefully get SSH anywhere the ocean exists on the r05 grid. But please let me know if you'd like to try something else

Reply:
@jonbob , thanks very much! I wonder if it is possible to map the MPAS-O SSH to all ocean cells. The ocean cells are identified in the land domain file. This may include extrapolations for the r05 cells that are identified as ocean by the land domain file but do not have overlap with the EC30to60E2r2 ocean grid. 
Otherwise, it may be difficult to tell a MOSART outlet where the nearest SSH is, if its nearest ocean cell does not have a remapped SSH. Please let me know if my description is not clear.

Reply:
Thanks for testing that, @fdongyu. And I think I understand what you're asking. It is tricky because remapping files in general apply only to the overlap between grids, and as you noticed the mpas-o grid may not correspond exactly with the inverse of the lnd domain in the r05 mosart file. Let me try a couple of things and see if I can come up with something else.

Reply:
@fdongyu  - Well, I have one other file for you to try -- but it's just a nearest-neighbor mapping created with ncremap:
````
/lcrc/group/e3sm/data/inputdata/cpl/cpl6/map_EC30to60E2r2_to_r05_neareststod.220728.nc
````
I'll ask someone else for any ideas, and if we don't come up with anything else your tool maybe the best answer. Thanks for your patience

Reply:
Thanks, @jonbob. I think this nearest-neighbor mapping works in a similar way as my approach, except that all land cells get a remapped ssh (see the remapped ssh below). I think this is fine since the ssh of the land cells will not be used. Please let me know if you have an updated solution. If not, I am fine with proceeding with this mapping file.  
![remapped_ssh](https://user-images.githubusercontent.com/10620053/181650182-1b412463-0407-424e-be09-c29c89fc198a.png)

Reply:
If it turns out we're going to use that python script for making maps, @fdongyu you can create your own repository with it and we can add it as an external module to mosart.   Or you can add it directly to the E3SM code (in components/mosart/utils) and put any DOI or Copyright info in a comment at the top of the file.

Reply:
@rljacob as another possibility, what do you think of https://github.com/E3SM-Project/E3SM-init?

Reply:
@fdongyu - I do have another file for you to try, but anvil/chrysalis seem unusable right now. Are you set up to test on any other platforms?

Reply:
@jonbob, thanks! Could you try Compy? All of my developments and tests are set up on Compy. I cannot access anvil/chrysalis and asked someone else to download the file for me.

Reply:
I have the file on compy, so that's easy:
/compyfs/inputdata/cpl/cpl6/map_EC30to60E2r2_to_r05_bilin.220728.nc

It's a typical bilinear remapping but uses a new(ish) extrapolation feature. I'm not sure it will be what you're looking for, but it's the only other thing I know of to try

Reply:
@jonbob, this file results in weird remapped ssh (figure below). But I think this nearest neighbor remapping file you generated  (`/lcrc/group/e3sm/data/inputdata/cpl/cpl6/map_EC30to60E2r2_to_r05_neareststod.220728.nc` )
should be used because its only difference from my approach is that the land cells are not excluded from the remapping. This is OK because the ssh on the land cells will never be used. Only the ssh in the river outlet cells will be used, whose value comes from the nearest ocean cell.  

If it is OK, I would like to proceed with your file. My script could be an option in the future after more tests (maybe with the unified land/ocean mesh) in the ICoM group.  What do you think? 

![remapped_ssh_bilinear](https://user-images.githubusercontent.com/10620053/182255189-3435f5ee-601a-496d-8de5-e8f40c3648b4.png)

Reply:
> @rljacob as another possibility, what do you think of https://github.com/E3SM-Project/E3SM-init?

Yes that's another possibility.

Reply:
@fdongyu - that sounds good. I didn't have much faith in this last extrapolated mapping, so let's go forward with the ncremap neareststod approach. But please do keep your script ready in case we find issues with the single-point nearest neighbor

Reply:
@jonbob, thanks. The mapping file is now updated. The result does not show much difference compared with that using my previous mapping file.